ttnn/cpp/ttnn/tensor/tensor.hpp

```diff
@@ -198,20 +201,18 @@ struct Tensor {
     // Non-Blocking Getters. Query attributes directly, without waiting for worker completion
     // ======================================================================================
     inline const Storage &storage() const { return this->tensor_attributes->storage; };
-    inline const tt::tt_metal::LegacyShape &legacy_shape() const { return this->tensor_attributes->shape.value; };
-    inline const ttnn::Shape &shape() const { return this->tensor_attributes->shape; };
-    inline const DataType &dtype() const { return this->tensor_attributes->dtype; };
-    inline const Layout &layout() const { return this->tensor_attributes->layout; };
-    inline const Tile &tile() const { return this->tensor_attributes->tile; };
+    inline tt::tt_metal::LegacyShape legacy_shape() const { return this->tensor_attributes->tensor_spec.compute_shape().value; };
+    inline ttnn::Shape shape() const { return this->tensor_attributes->tensor_spec.compute_shape(); };
+    inline DataType dtype() const { return this->tensor_attributes->tensor_spec.tensor_layout().get_data_type(); };
+    inline Layout layout() const { return this->tensor_attributes->tensor_spec.tensor_layout().get_layout(); };
+    inline Tile tile() const { return this->tensor_attributes->tensor_spec.tensor_layout().get_page_config().get_tile(); };
```

Comment:
lets discuss this. ideally we should not calculate these all the time, they are not changing dynamically

Reply:
The only calculated one out of theses getters is `shape()`. I'm planning to look into how often each `compute_*` method is called and cache some of them.